ydb/library/yql/providers/pq/gateway/dummy/yql_pq_file_topic_client.cpp

```diff
@@ -0,0 +1,293 @@
+        return NThreading::Async([this] () {
+            EventsQ_.BlockUntilEvent();
+            return NThreading::MakeFuture();
+        }, Pool_);
+    }
+
+    TVector<NYdb::NTopic::TReadSessionEvent::TEvent> GetEvents(bool block, TMaybe<size_t> maxEventsCount, size_t maxByteSize) override {
+        // TODO
+        Y_UNUSED(maxByteSize);
+
+        TVector<NYdb::NTopic::TReadSessionEvent::TEvent> res;
+        for (auto event = EventsQ_.Pop(block); !event.Empty() &&  res.size() <= maxEventsCount; event = EventsQ_.Pop(/*block=*/ false)) {
+            res.push_back(*event);
+        }
+        return res;
+    }
+
+    TVector<NYdb::NTopic::TReadSessionEvent::TEvent> GetEvents(const NYdb::NTopic::TReadSessionGetEventSettings& settings) override {
+        return GetEvents(settings.Block_, settings.MaxEventsCount_, settings.MaxByteSize_);
+    }
+
+    TMaybe<NYdb::NTopic::TReadSessionEvent::TEvent> GetEvent(bool block, size_t maxByteSize) override {
+        // TODO
+        Y_UNUSED(maxByteSize);
+
+        return EventsQ_.Pop(block);
+    }
+
+    TMaybe<NYdb::NTopic::TReadSessionEvent::TEvent> GetEvent(const NYdb::NTopic::TReadSessionGetEventSettings& settings) override {
+        return GetEvent(settings.Block_, settings.MaxByteSize_);
+    }
+
+    bool Close(TDuration timeout = TDuration::Max()) override {
+        Y_UNUSED(timeout);
+        // TOOD send TSessionClosedEvent
+        EventsQ_.Stop();
+        FilePoller_.join();
+        return true;
+    }
+
+    NYdb::NTopic::TReaderCounters::TPtr GetCounters() const override {
+        return Counters_;
+    }
+
+    TString GetSessionId() const override {
+        return ToString(Session_->GetPartitionSessionId());
+    }
+
+    ~TFileTopicReadSession() {
+        EventsQ_.Stop();
+        FilePoller_.join();
+    }
+
+private:
+    using TMessageInformation = NYdb::NTopic::TReadSessionEvent::TDataReceivedEvent::TMessageInformation; 
+    using TMessage = NYdb::NTopic::TReadSessionEvent::TDataReceivedEvent::TMessage; 
+
+    TMessageInformation MakeNextMessageInformation(size_t offset, size_t uncompressedSize, const TString& messageGroupId = "") { 
+        auto now = TInstant::Now(); 
+        TMessageInformation msgInfo(
+            offset,
+            ProducerId_,
+            SeqNo_,
+            now,
+            now,
+            MakeIntrusive<NYdb::NTopic::TWriteSessionMeta>(),
+            MakeIntrusive<NYdb::NTopic::TMessageMeta>(),
+            uncompressedSize,
+            messageGroupId
+        );
+        return msgInfo;
+    }
+    
+    TMessage MakeNextMessage(const TString& msgBuff) {
+        TMessage msg(msgBuff, nullptr, MakeNextMessageInformation(MsgOffset_, msgBuff.size()), Session_);
+        return msg;
+    }
+
+    void PollFileForChanges() {
+        TFileInput fi(File_);
+        while (!EventsQ_.IsStopped()) {
+            TString rawMsg;
+            TVector<TMessage> msgs;
+
+            while (size_t read = fi.ReadLine(rawMsg)) {
```

Comment:
What will happen if file is half-written and ends in the middle of line?
From brief look at ReadLine implementation (`DoReadTo`), I'm under impression it will be broken (one call `ReadLine` will return incomplete line, then file size change will be detected and next call to `ReadLine` will return second half of line).

Reply:
You can simulate this with
```sh
echo -n foo > stream
sleep 10s
echo bar >> stream
```
This should return single `foobar` event, I expect result will be two events -- `foo` and `bar`

Reply:
User is supposed to write entire line at once with assumption that such write is atomic